### ISSUE TITLE
Dimension mismatch between data and custom-trained acoustic model

### ISSUE THREAD

**Issue by nrgslp** [MontrealCorpusTools/Montreal-Forced-Aligner#225]
Thanks for your help in getting this far! I don't quite have my head wrapped around what's going on for this one...

Command:
` mfa train_dictionary /mnt/c/.../Rhotic-Corpus-zCondor-sync/gt-rhotics/ /mnt/c/.../Rhotic-Corpus-zExtras/MFA/gt-r_phonetic_fortraining.dict /home/.../Documents/gt-rhotics-out/model_phonOrthodict.zip /home/.../Documents/gt-rhotics-out/traineddict.dict`

metal.yaml file for the acoustic models:
```
architecture: gmm-hmm
features:
  deltas: false
  fmllr: true
  frame_shift: 10
  lda: true
  pitch: false
  splice_left_context: null
  splice_right_context: null
  type: mfcc
  use_energy: false
```

**Error in `align.final.0.log`:**
`ERROR (gmm-align-compiled[5.5]:LogLikelihoodZeroBased():gmm/decodable-am-diag-gmm.cc:50) Dim mismatch: data dim = 39 vs. model dim = 40`

I'm unsure if this is a bug - or if this is something that I can account for through a custom configuration file during model training? (And, if so, what to change in the config file.)

I found something that may be relevant in the [Kaldi boards](https://sourceforge.net/p/kaldi/discussion/1355348/thread/028ffd7f/), but I'm unsure of what to do with the info:

> Actually, since the data dim is 39 and the model dim is 40, I think it's most likely that he is using delta+accel features but the model expected LDA+MLLT features.

Thanks for any thoughts!

**Comment by mmcauliffe** [comment]
Yeah, exactly, this is probably a bug in the alignment code where it's not properly extracting the lda.mat file and using it to generate features. I'll take a look tonight!

**Comment by nrgslp** [comment]
This makes sense - I knew of the 39 features for the MFCCs, deltas, and delta-deltas, and I appreciate the walk-through for the LDA features. 

For the record I am using 2.0.0.a2 here. Let me try again with 2.0.0.a4.

nrg

**Comment by mmcauliffe** [comment]
that warning shouldn't matter too much in that particular area, it's just trying to get the feature dimension, so I think it broke early because of that, but it would have thrown a bigger error if it didn't get the correct dimension

**Comment by nrgslp** [comment]
Okay, re-running everything on 2.0.0.a4 seemed to work. My apologies for raising a non-issue!

**Comment by mmcauliffe** [comment]
Not a problem at all, the LDA training was non-functional before 2.0.0.a4, so that's on me ultimately!
